RELEASE_NOTES.md

```diff
@@ -3,6 +3,7 @@
 ## in develop
 
 * Handles empty scatters in WDL workflows
+* Reserved parameters are now placed in the "Reserved" parameter group (only affects the display of the app/workflow in the UI)
```

Comment:
Should we name the group something that is very clear to the user that it's something system-like, set by `dxCompiler`? E.g. `Reserved for dxCompiler (do not change)` or just `Reserved for dxCompiler`.